src/Common/formatIPv6.cpp

```diff
@@ -165,9 +165,9 @@ void formatIPv6(const unsigned char * src, char *& dst, uint8_t zeroed_tail_byte
             uint8_t ipv4_buffer[IPV4_BINARY_LENGTH] = {0};
             memcpy(ipv4_buffer, src + 12, IPV4_BINARY_LENGTH);
             // Due to historical reasons formatIPv4() takes ipv4 in BE format, but inside ipv6 we store it in LE-format.
-            if constexpr (std::endian::native == std::endian::little)
-                std::reverse(std::begin(ipv4_buffer), std::end(ipv4_buffer));
-
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
Unless there are good reasons for the C-style macro form, I'd prefer the C++-native `std::endian`.

Reply:
native style doesn't compile on s390x platform. error: code will never be executed [-Werror,-Wunreachable-code]

Reply:
Hm, okay. Perhaps an issue with the compiler on s390x.